src/watcher/watcher.ts

```diff
@@ -14,171 +14,176 @@ const log = pog.sub('cli:dev:watcher')
 /**
  * Entrypoint into the watcher system.
  */
-export function createWatcher(opts: Opts) {
-  const cfg = cfgFactory(opts)
-
-  compiler.init(opts)
-  compiler.stop = stopRunner
-
-  // Run ./dedupe.js as preload script
-  if (cfg.dedupe) process.env.NODE_DEV_PRELOAD = __dirname + '/dedupe'
-
-  //
-  // Setup State
-  //
-
-  // Create a file watcher
-
-  // TODO watch for changes to tsconfig and take correct action
-  // TODO watch for changes to package json and take correct action (imagine
-  // there could be graphql-santa config in there)
-  // TODO restart should take place following npm install/remove yarn
-  // add/remove/install etc.
-  // TODO need a way to test file matching given patterns. Hard to get right,
-  // right now, and feedback loop sucks. For instance allow to find prisma
-  // schema anywhere except in migrations ignore it, that is hard right now.
-
-  const pluginWatchContributions = opts.plugins.reduce(
-    (patterns, p) =>
-      patterns.concat(p.dev.addToSettings.watchFilePatterns || []),
-    [] as string[]
-  )
-
-  const pluginIgnoreContributions = opts.plugins.reduce(
-    (patterns, p) =>
-      patterns.concat(p.dev.addToSettings.ignoreFilePatterns || []),
-    [] as string[]
-  )
-
-  const watcher = watch([opts.layout.sourceRoot, ...pluginWatchContributions], {
-    ignored: ['./node_modules', './.*', ...pluginIgnoreContributions],
-    ignoreInitial: true,
-    onAll(event, file) {
-      for (const p of opts.plugins) {
-        p.dev.onFileWatcherEvent?.(event, file)
+export function createWatcher(opts: Opts): Promise<void> {
+  return new Promise((resolve, reject) => {
```

Comment:
Main change is this promise wrapper and the `process.on`, everything else is noise.